docs(U-Boot): Refactor K3 and OMAP Build Guide

jonaswood01 opened the pull request:
Mostly cleanup and organization. Some content updated on OMAP. Turns large build guide into its own section under U-Boot, with 6 pages under the build guide. Give feedback if content needs more updating, and I will address in future PR.

cshilwant reviewed (changes requested):
Resolve merge conflicts and failures reported by 3 workflows

sadik-smd commented on source/devices/AM57X/linux/Release_Specific_Release_Notes.rst:
```diff
@@ -99,6 +99,7 @@ See :ref:`release-specific-supported-platforms-and-versions` for a list of suppo
+.. _release-specific-build-information:
 
 Build Information
 =====================================
```
Should be aligned with text

jonaswood01 replied:
@sadik-smd not sure what you're suggesting here. Alignment follows internal link convention inside this and other rst files in PSDK docs, as well as Sphinx RST internal link convention: https://www.sphinx-doc.org/en/master/usage/referencing.html#ref-role

StaticRocket commented on source/linux/Foundational_Components/U-Boot/BG-Setup-OMAP.rst:
```diff
@@ -0,0 +1,49 @@
+
+.. _Getting the U-Boot Source Code-label:
+
+^^^^^^^^^^^^^^^^^^^^^^^
+Getting the Source Code
+^^^^^^^^^^^^^^^^^^^^^^^
```
https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections

jonaswood01 replied:
Changed instances of section headers to have correct order in the 12 u-boot build guide files per the Sphinx link: '#' for beginning of file, then '*', then '=', etc.

StaticRocket commented on source/linux/Foundational_Components/U-Boot/BG-Environment-K3.rst:
```diff
@@ -0,0 +1,100 @@
+Environment
+***********
+
+^^^^^^^^^^^^^^^^^^
+U-Boot Environment
+^^^^^^^^^^^^^^^^^^
```
https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections

StaticRocket commented on source/linux/Foundational_Components/U-Boot/BG-Build-OMAP.rst:
```diff
@@ -0,0 +1,151 @@
+| K2G ICE                    | ``k2g_evm_defconfig``       |                          |                                              |                          |                          |                          |                                         |                                          |
++----------------------------+-----------------------------+--------------------------+----------------------------------------------+--------------------------+--------------------------+--------------------------+-----------------------------------------+------------------------------------------+
+| OMAP-L138 LCDK             | ``omapl138_lcdk_defconfig`` |                          | ``omapl138_lcdk_defconfig``                  |                          |                          |                          |                                         |                                          |
++----------------------------+-----------------------------+--------------------------+----------------------------------------------+--------------------------+--------------------------+--------------------------+-----------------------------------------+------------------------------------------+
```
This would be much easier to review as a list-table

https://docutils.sourceforge.io/docs/ref/rst/directives.html#list-table

jonaswood01 replied:
You're right, but this is just too big a revamp for this PR, but can consider for future PR. I'd need to update the same tables in `BG-Build-K3.rst`

StaticRocket commented on source/linux/Foundational_Components/U-Boot/BG-Build-OMAP.rst:
```diff
@@ -0,0 +1,151 @@
+
+.. ifconfig:: CONFIG_part_variant in ('AM335X')
+
+   If you used 'O=am335x\_evm' as your object directory:
```
This is also applicable to other parameters listed here

```suggestion
   If you used ``O=am335x_evm`` as your object directory:
```

StaticRocket commented on source/linux/Foundational_Components/U-Boot/Users-Guide.rst:
```diff
@@ -6,29 +6,28 @@ User's Guide
-    :maxdepth: 1
+   :maxdepth: 1
 
-    UG-General-Info
```
You removed ``UG-General-Info``, but left unreachable references in:
```
configs/AM69/AM69_linux_toc.txt
configs/AM68/AM68_linux_toc.txt
configs/J721S2/J721S2_linux_toc.txt
configs/TDA4VM/TDA4VM_linux_toc.txt
configs/AM67/AM67_linux_toc.txt
configs/J7200/J7200_linux_toc.txt
configs/GEN/GEN_linux_toc.txt
configs/J742S2/J742S2_linux_toc.txt
configs/AM68A/AM68A_linux_toc.txt
configs/DRA821A/DRA821A_linux_toc.txt
configs/AM67A/AM67A_linux_toc.txt
configs/AM69A/AM69A_linux_toc.txt
configs/J784S4/J784S4_linux_toc.txt
configs/J722S/J722S_linux_toc.txt
configs/J721E/J721E_linux_toc.txt
```

jonaswood01 commented:
Changed commit tag to "refactor". Addressed all feedback, resolved merge conflicts, and passed all checks except Vale. I reviewed output from Vale and felt no spelling or grammar changes are necessary, leaving as-is. Please re-review.

StaticRocket reviewed (dismissed):
Considering this is primarily a refactor splitting the doc we shouldn't be too concerned about vale right now. Other warnings are still valid.

cshilwant reviewed (changes requested):
Resolve merge conflicts 

> source/linux/How_to_Guides/Target/How_to_boot_quickly.rst

StaticRocket commented:
Ah, needs to be rebased now

jonaswood01 commented on source/devices/AM62AX/linux/Release_Specific_Release_Notes.rst:
```diff
@@ -49,7 +49,7 @@ What's new
   - EdgeAI memory carveouts now supported via `k3-am62a7-sk-edgeai.dtso <https://git.ti.com/cgit/ti-linux-kernel/ti-linux-kernel/tree/arch/arm64/boot/dts/ti/k3-am62a7-sk-edgeai.dtso?h=11.01.07>`_ in ti-linux-kernel & applied by default in the AM62A board environment via the name_overlays variable in ti-u-boot as seen in `board/ti/am62ax/am62ax.env <https://git.ti.com/cgit/ti-u-boot/ti-u-boot/tree ...
   - Support for SELinux via meta-selinux with tisdk-edgeai-image - :ref:`Building the SELinux image with Yocto <building-the-sdk-with-yocto>` and :ref:`SELinux oeconfig file <yocto-layer-configuration>`
-  - EdgeAI DM R5 (:file:`dm_edgeai_mcu1_0_release_strip.out`) & C7x IPC (:file:`dsp_edgeai_c7x_1_release_strip.out`) firmwares delivered via `ti-linux-firmware <https://git.ti.com/cgit/processor-firmware/ti-linux-firmware/tree/?h=11.01.07>`__
+  - EdgeAI DM R5 (:file:`dm_edgeai_mcu1_0_release_strip.out`) & C7x IPC (:file:`dsp_edgeai_c7x_1_release_strip.out`) firmwares delivered via `ti-linux-firmware v11.01.07 <https://git.ti.com/cgit/processor-firmware/ti-linux-firmware/tree/?h=11.01.07>`__
```
@cshilwant upon latest rebase, I again had to change `ti-linux-firmware` to `ti-linux-firmware v11.01.07` here to resolve `rstcheck` warning: `source/devices/AM62AX/linux/Release_Specific_Release_Notes.rst:131: (INFO/1) Duplicate implicit target name: "ti-linux-firmware".` Please review.
